jwst/cube_build/src/cube_match_sky.c

```diff
@@ -0,0 +1,1441 @@
+int overlap_slice_with_spaxels(int overlap_partial,
+			       double cdelt1, double cdelt2,
+			       int naxis1, int naxis2,
+			       double xcenters[], double ycenters[],
+			       double xi_min, double eta_min,
+			       double xi_max, double eta_max,
+			       int wave_slice_dq[]) {
+
+  /* 
+     Set the initial dq plane indicating if the input data falls on a spaxel
+
+     This algorithm assumes the input data falls on a line in the IFU cube, which is
+     the case for NIRSpec slices. The NIRSpec slice's endpoints are used to determine
+     which IFU spaxels the slice falls on to set an initial dq flag.
+     
+     Parameters
+     ---------
+     overlap_partial: intermediate dq flag
+
+     wavelength: the wavelength bin of the IFU cube working with
+
+     Sets
+     ----
+     wave_slice_dq : array containing intermediate dq flag
+
+     Bresenham's Line Algorithm to find points a line intersects with grid.
+
+     Given the endpoints of a line find the spaxels this line intersects.
+
+     Returns
+     -------
+     Points: a tuple of x,y spaxel values that this line intersects
+
+  */
+
+  int status = 0;
+
+  //set up line - convert to integer values
+  int x1 = (int)((xi_min - xcenters[0]) / cdelt1);
+  int y1 = (int)((eta_min - ycenters[0]) / cdelt2);
+  int x2 = (int)((xi_max - xcenters[0]) / cdelt1);
+  int y2 = (int)((eta_max - ycenters[0]) / cdelt2);
+
+  int dx = x2 - x1;
+  int dy = y2 - y1;
+
+  bool is_steep;
+  is_steep = abs(dy) > abs(dx);
+
+  // if is_steep switch x and y 
+  if (is_steep){
+      x1 = y1;
+      y1 = x1;
+      x2 = y2;
+      y2 = x2;
+    }
+
+  // Swap start and end points if necessary and store swap state
+  //bool swapped;
+  //swapped = false;
+  if (x1 > x2){
+    x1 = x2;
+    x2 = x1;
+
+    y1 = y2;
+    y2 = y1;
+    //swapped = true;
+  }
+
+  // Recalculate differences
+  dx = x2 - x1;
+  dy = y2 - y1;
+
+  //calculate error
+  int error = (int)(dx / 2.0);
+  int ystep = -1;
+  if (y1 < y2){
+    ystep = 1;
+  }
+
+  // iterate over grid to generate points between the start and end of line
+  int y = y1;
+  for (int x = x1; x< (x2 + 1); x++){
+    int yuse  = y;
+    int xuse = x ;
+    if (is_steep){
+	yuse = x;
+	xuse = y;
+      }
+    //coord = (y, x) if is_steep else (x, y);
+    int index = (yuse * naxis1) + xuse;
+    wave_slice_dq[index] = overlap_partial;
+    error -= abs(dy);
+    if (error < 0){
+      y += ystep;
+      error += dx;
+    }
+
+  }
+  return status; 
+}
+
+//________________________________________________________________________________
+// Set the spaxel dq = 0. This is used when not determining the FOV on the sky for
+// setting the DQ plane. This is case for internalCal type cubes
+
+int dq_set_zero(int ncube, int **spaxel_dq){
+    int *idqv = NULL;  // int vector for spaxel
+
+    if (mem_alloc_dq(ncube, &idqv)) return 1;
+
+    // Set all data to zero
+    for (long i = 0; i < ncube; i++){
+      idqv[i] = 0;
+    }
+     *spaxel_dq = idqv;
```

Comment:
Assuming you have switched to `calloc` as suggested above
```suggestion
    if (mem_alloc_dq(ncube, spaxel_dq)) return 1;
```